tiledb/sm/filesystem/uri.h

```diff
@@ -242,6 +242,17 @@ class URI {
   Status get_rest_components(
       std::string* array_namespace, std::string* array_uri) const;
 
+  /**
+   * Return the fragment name from the URI if one can be found.
+   *
+   * The logic for this parsing is that first we locate a '/__fragments/' path
+   * component in the string and then take everything up to the next possibly
+   * non-existent '/' separator.
+   *
+   * @return The fragment name URI if one is found, else std::nullopt.
+   */
+  std::optional<URI> get_fragment_name() const;
```

Comment:
While I would prefer this in `storage_format`, this is adequate sequestration of the code for now.